public/js/utils.js

```diff
@@ -0,0 +1,88 @@
+async function enableEthereum() {
+  if (window.ethereum) {
+    metamask = new Web3(window.ethereum);
+    try {
+      await window.ethereum.enable();
+      console.log('ethereum enabled...');
+    } catch(error) {
+      console.error(error);
+    }
+  }
+  else if (window.web3) {
+    window.web3 = new Web3(window.web3.currentProvider);
+    console.log('Legacy web3 detected!');
+  }
+  else {
+    console.log('Non-Ethereum browser detected!');
+    return;
+  }
+}
+
+
+function buf2hex(buffer) { // buffer is an ArrayBuffer
+  return Array.prototype.map.call(new Uint8Array(buffer), x => ('00' + x.toString(16)).slice(-2)).join('');
+}
+
+function submitTransaction() {
+  var wastCode = document.getElementById('wastCode').value;
+  var value = document.getElementById('value').value;
+
+  var wast = "";
+  var wasm = "";
+
+  // Validate wast
+  if (wastCode.length > 0) {
+
+    // Compile wast to wasm
+    try {
+      var module = window.Binaryen.parseText(wastCode);
+      wasm = buf2hex(module.emitBinary());
+    } catch (e) {
+      alert(e);
+    }
+
+    // format hex to "wasm data"
+    for (var i = 0; i < wasm.length; i +=2) {
+      wast += "\\" + wasm.slice(i, i + 2);
+    }
+
+    // wrap wasm data into a deployer wast contract
+    wast = `(module (import "ethereum" "finish" (func $finish (param i32 i32))) (memory 100) (data (i32.const 0)  "${wast}") (export "memory" (memory 0)) (export "main" (func $main)) (func $main (call $finish (i32.const 0) (i32.const ${wasm.length / 2}))))`;
```

Comment:
`(memory 100)` seems to be invalid. It should actually be set to `wasm.length / 65536`.